Instagram/Instagram/Global/DesignSystem/UIComponent/IDSButton.swift

```diff
@@ -0,0 +1,119 @@
+//
+//  IDSButton.swift
+//  Instagram
+//
+//  Created by 소연 on 2022/04/04.
+//
+
+import UIKit
+
+import SnapKit
+import Then
+
+final class IDSButton: UIButton {
+    
+    enum FontWeight {
+        case regular
+        case medium
+        case bold
+        case semiBold
+    }
+    
+    // MARK: - Properties
+    
+    private var leftIconImageView = UIImageView().then {
+        $0.contentMode = .scaleAspectFit
+        $0.isHidden = true
+    }
+    
+    private var rightIconImageView = UIImageView().then {
+        $0.contentMode = .scaleAspectFit
+        $0.isHidden = true
+    }
+    
+    var isActivated: Bool = false {
+        didSet {
+            self.backgroundColor = self.isActivated ? activatedBackgroundColor : normalBackgroundColor
+            self.setTitleColor(self.isActivated ? activatedFontColor : normalFontColor, for: .normal)
+            self.isEnabled = isActivated
+        }
+    }
+    
+    private var normalBackgroundColor: UIColor = .lightGray
+    private var normalFontColor: UIColor = .darkGray
+    
+    private var activatedBackgroundColor: UIColor = .systemBlue
+    private var activatedFontColor: UIColor = .white
+    
+    init() {
+        super.init(frame: .zero)
+        configUI()
+        setLayout()
+    }
+    
+    required init(coder aDecoder: NSCoder) {
+        super.init(coder: aDecoder)!
+        configUI()
+        setLayout()
+    }
+    
+    // MARK: - Init UI
+    
+    private func configUI() {
+        layer.cornerRadius = 5
+    }
+    
+    private func setLayout() {
+        
+    }
+    
+    // MARK: - Custom Method
+    
+    internal func setButtonColors(normalBackgroundColor: UIColor,
+                             normalFontColor: UIColor,
+                                  activatedBackgroundColor: UIColor,
+                             activatedFontColor: UIColor) {
+        self.normalBackgroundColor = normalBackgroundColor
+        self.normalFontColor = normalFontColor
+        self.activatedBackgroundColor = activatedBackgroundColor
+        self.activatedFontColor = activatedFontColor
+    }
+    
+    internal func setTitleWithStyle(title: String,
+                                  size: CGFloat,
+                                  weight: FontWeight = .regular) {
+        let font: UIFont
+        
+        // FIXME: - 폰트 수정
+        switch weight {
+        case .regular:
+            font = IDSFont.body1
+        case .medium:
+            font = IDSFont.body1
+        case .bold:
+            font = IDSFont.body1
+        case .semiBold:
+            font = IDSFont.body1
+        }
+        
+        self.titleLabel?.font = font
+        self.setTitle(title, for: .normal)
+    }
+    
+    internal func setLeftIcon(imageName: String) {
+        leftIconImageView.isHidden = false
+        leftIconImageView.image = UIImage(named: imageName)
+    }
```

Comment:
이건 그냥 든 생각인데, parameter로 String을 받는것도 좋지만 지금 프로젝트에서 SwiftGen을 사용하고 있으니까 바로 자동완성될 수 있게 uiImage 타입을 넣으면 어떨까?

```suggestion
    internal func setLeftIcon(image: UIImage) {
        leftIconImageView.isHidden = false
        leftIconImageView.image = image
    }
```

Reply:
너무 좋다고 생각 .. 습관을 버리지 못한 ... 저에게 참언을 해주셔서 감사합니다 역시 최이준 최고 👍🏻